src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -139,35 +140,54 @@ internal PEReader PEReaderOpt
             }
         }
 
-        internal unsafe MetadataReader MetadataReader
+        internal MetadataReader MetadataReader
         {
             get
             {
                 if (_lazyMetadataReader == null)
                 {
-                    MetadataReader newReader;
+                    InitializeMetadataReader();
+                }
+                else if (_isDisposed)
```

Comment:
Oh, I see what you mean. Actually the fix is guarding against your pattern as well. ```m.MetadataReader``` goes thru the same code path.